pkg/api/v1alpha1/notifications_target.go

```diff
@@ -0,0 +1,462 @@
+		}
+
+		sendError(c, http.StatusBadRequest, msg)
+
+		return
+	}
+
+	event, err := dbtools.AuditNotificationTargetDeleted(
+		c.Request.Context(),
+		tx,
+		getCtxAuditID(c),
+		getCtxUser(c),
+		n,
+	)
+	if err != nil {
+		msg := fmt.Sprintf("error deleting notification target (audit): %s", err.Error())
+
+		if err := tx.Rollback(); err != nil {
+			msg += fmt.Sprintf("error rolling back transaction: %s", err.Error())
+		}
+
+		sendError(c, http.StatusBadRequest, msg)
+
+		return
+	}
+
+	if err := updateContextWithAuditEventData(c, event); err != nil {
+		msg := fmt.Sprintf("error deleting notification target: %s", err.Error())
+
+		if err := tx.Rollback(); err != nil {
+			msg += fmt.Sprintf("error rolling back transaction: %s", err.Error())
+		}
+
+		sendError(c, http.StatusBadRequest, msg)
+
+		return
+	}
+
+	if err := tx.Commit(); err != nil {
+		msg := fmt.Sprintf("error committing notification target delete: %s", err.Error())
+
+		if err := tx.Rollback(); err != nil {
+			msg += fmt.Sprintf("error rolling back transaction: %s", err.Error())
+		}
+
+		sendError(c, http.StatusBadRequest, msg)
+
+		return
+	}
+
+	if err := dbtools.RefreshNotificationDefaults(c.Request.Context(), r.DB); err != nil {
+		sendError(c, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	err = r.EventBus.Publish(
+		c.Request.Context(),
+		events.GovernorNotificationTargetsEventSubject,
+		&events.Event{
+			Version:              events.Version,
+			Action:               events.GovernorEventDelete,
+			AuditID:              c.GetString(ginaudit.AuditIDContextKey),
+			ActorID:              getCtxActorID(c),
+			NotificationTargetID: n.ID,
+		},
+	)
+	if err != nil {
+		sendError(
+			c,
+			http.StatusBadRequest,
+			fmt.Sprintf(
+				"failed to publish notification target delete event: %s\n%s",
+				err.Error(),
+				"downstream changes may be delayed",
+			),
+		)
+
+		return
+	}
+
+	c.JSON(http.StatusAccepted, n)
+}
+
+// updateNotificationTarget updates a notification target in DB
+func (r *Router) updateNotificationTarget(c *gin.Context) {
+	id := c.Param("id")
+
+	q := qm.Where("id = ?", id)
+	if _, err := uuid.Parse(id); err != nil {
+		q = qm.Where("slug = ?", id)
+	}
+
+	n, err := models.NotificationTargets(q).One(c.Request.Context(), r.DB)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			sendError(c, http.StatusNotFound, "notification target not found"+err.Error())
+			return
+		}
+
+		sendError(c, http.StatusInternalServerError, "error getting notification target: "+err.Error())
+
+		return
+	}
+
+	original := *n
+
+	req := &NotificationTargetReq{}
+	if err := c.BindJSON(req); err != nil {
+		sendError(c, http.StatusBadRequest, "unable to bind request: "+err.Error())
+		return
+	}
+
+	if req.Name != "" {
+		sendError(c, http.StatusBadRequest, "modifying notification target name is not allowed")
+		return
+	}
+
+	if req.Description != "" {
+		n.Description = req.Description
+	}
+
+	n.DefaultEnabled = req.DefaultEnabled
```

Comment:
doesn't this mean the field is always required? what if its true but I don't pass it bc I don't want it to change (like the other fields)?